docs/HowToFAQ/Errors-CommandLine.md

```diff
@@ -1,24 +1,231 @@
 
-<!-- TODO: Errors in PrintMode? -->
-<!-- TODO: Errors in diagnosticsFormat? -->
+## **dafny: Error: Invalid argument _argument_ to option print-mode_help_** {#cli_print_mode}
+
+<!-- %check-cli-err -->
+```bash 
+dafny -printMode:zzz
+```
+
+The `printMode` option has a number of alternative values that are spelled out in the help document (`dafny -?`).
+It is also recommended to use the new CLI, with the option `--print-mode`.
 
-## Error: Invalid argument _argument_ to option _option_
+## Unsupported diagnostic format: _format_; expecting one of 'json', 'text'.** {#cli_diagnostic_format}
 
 <!-- %check-cli -->
+```bash
+dafny /diagnosticsFormat:zzz
+```
+
+This option controls the format of error messages. Typically the 'text' format is used (and is the default).
+But 'json' is also an option. Any other choice is illegal and results in this error message.
+
+## **dafny: Error: Invalid argument _argument_ to option functionSyntax_help_** {#cli_function_syntax}
+
+<!-- %check-cli-err -->
 ```bash 
-dafny resolve --function-syntax:2 mod.dfy
+dafny -functionSyntax:z
 ```
 
-This is a generic error message about command-line arguments,
-indicating that the value given (after the `:` or `=` or as the next command-line argument) is not recognized as a valid value for the given option.
+The `functionSynax` option has a number of alternative values that are spelled out in the help document (`dafny -?`),
+most commonly the values '3' and '4' ('4' for dafny version 4.0 and later).
+It is also recommended to use the new CLI, with the option `--function-syntax`.
 
-The correct spellings of the valid values should be stated 
-in the help document (`dafny -?` or `dafny <command> -h`).
+## **dafny: Error: Invalid argument _argument_ to option quantifierSyntax_help_** {#cli_quantifier_syntax}
 
-## Error: Invalid argument _argument_ to option --quantifier-syntax
+<!-- %check-cli-err -->
+```bash 
+dafny -quantifierSyntax:z
+```
 
-<!-- %check-cli -->
+The `quantifierSyntax ` option has a number of alternative values that are spelled out in the help document (`dafny -?`),
+most commonly the values '3' and '4' ('4' for dafny version 4.0 and later).
+It is also recommended to use the new CLI, with the option `--quantifier-syntax`.
+
+## **dafny: Error: Invalid argument _argument_ to option verificationLogger _help_** {#cli_verification_logger}
+
+<!-- %check-cli-err -->
+```bash 
+dafny -verificationLogger:z
+```
+
+The `verificationLogger` option has these alternatives: trx, csv, text.
+The option name in the new cli is `--log-format`.
+
+## **dafny: Error: Invalid argument _argument_ to option testContracts_help_** {#cli_test_contracts}
+
+<!-- %check-cli-err -->
+```bash 
+dafny -testContracts:z
+```
+
+The `testContracts` option has these alternatives: Externs, TestedExterns.
+The similar option in the new cli is `--test-assumptions`.
+
+## **dafny: Error: Invalid argument _argument_ to option printIncludes_help_** {#cli_print_includes}
```

Comment:
Is the `_help_` bit at the end of `option printIncludes_help_` intentional? (Here and in other headings)

Reply:
Yes - a bit unfortunate for readability I know. The italicized elements are placeholders for variable text in the error message. They are also converted into .* matchers when matching the error message in the heading to the actual message obtained by applying dafny to the code example, to check that everything is consistent.